csrc/host_ir/jit.cpp

```diff
@@ -42,15 +42,25 @@
 
 namespace nvfuser {
 
-using AllocateFunc = std::function<
-    void(const int64_t*, int64_t, const int64_t*, int64_t, at::Tensor&)>;
+using main_func_t = std::function<at::Tensor**(const at::Tensor**)>;
```

Comment:
Instead of Tensor**, can we flatten the input/output args? This way, we don't need to worry about allocations/deallocations of the `Tensor*[]`.

Reply:
Got it! I certainly missed that.